community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/slurmsync.py

```diff
@@ -152,6 +151,13 @@ def _find_dynamic_node_status() -> NodeAction:
     # * delete orhpaned instances
     return NodeActionUnchanged()  # don't touch dynamic nodes
 
+def get_fr_action(fr: FutureReservation, nodename:str, state:NodeState) -> Optional[NodeAction]:
+    now = datetime.utcnow()
+    if fr and now<fr.start_time:
+        return NodeActionDown(reason=f"Waiting for reservation:{fr.name} to start at {fr.start_time}") if state.base != "DOWN" else NodeActionUnchanged()
+
+    if fr and now>fr.end_time:
```

Comment:
`fr` is never `None`, otherwise it would be typed as `Optional[FutureReservation]` )

Reply:
Yes I forgot to update `node_is_fr` to return fr is it exists. Changed that so now we know it is never None